### PR TITLE
py-tensorflow-estimator: new versions

### PR DESCRIPTION
**Pull request by glennpj**
Add versions
- 2.5.0
- 2.6.0
- 2.7.0

Depends on the following PRs:
- #27138 
- #27991

### CODE REVIEW

**Comment by glennpj** [comment]
@spackbot run pipeline

**Review by adamjstewart — COMMENTED** [review]
Don't really know where to look in this package for a list of deps. Also, I don't see funcsigs anywhere with a recursive grep

**Comment by glennpj** [comment]
> Don't really know where to look in this package for a list of deps. Also, I don't see funcsigs anywhere with a recursive grep

Same here. I determined the py-keras dependencies from building. I left the py-funcsigs dependency as-is because I was not sure, but I can try some builds without it to see what shows up.

**Comment by glennpj** [comment]
The `py-funcsigs` package is needed for `python@:3.2`.

File: var/spack/repos/builtin/packages/py-tensorflow-estimator/package.py [hunk 1]
```diff
@@ -26,6 +29,11 @@ class PyTensorflowEstimator(Package):
 
     extends('python')
 
+    depends_on('py-keras@2.7.0:2.7', type=('build', 'run'), when='@2.7.0')
+    depends_on('py-keras@2.6.0:2.6', type=('build', 'run'), when='@2.6.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow-estimator/package.py:33]: I don't see any deps declared other than TF

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-tensorflow-estimator/package.py:33]: As I said in https://github.com/spack/spack/pull/28093#issuecomment-1002196071:

> I determined the py-keras dependencies from building.

**Review by adamjstewart — DISMISSED** [review]
Looks good as far as I can tell. The whole TF ecosystem is confusing and filled with circular deps. I wish they could just package software like normal developers.

**Comment by adamjstewart** [comment]
Can you resolve the merge conflicts?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**